ext/standard/array.c

```diff
@@ -6622,14 +6622,14 @@ static zend_result php_array_find(const HashTable *array, zend_fcall_info fci, z
 		if (!str_key) {
 			ZVAL_LONG(&args[1], num_key);
 		} else {
-			ZVAL_STR_COPY(&args[1], str_key);
+			ZVAL_STR(&args[1], str_key);
 		}
 
-		ZVAL_COPY(&args[0], operand);
+		ZVAL_COPY_VALUE(&args[0], operand);
 
 		zend_result result = zend_call_function(&fci, &fci_cache);
 		if (EXPECTED(result == SUCCESS)) {
```

Comment:
I will point out that I don't think it is possible for the call to fail in this case.
If you want to check an exception has been thrown one needs to check that the retval is `IS_UNDEF`

Reply:
Oh, that was a fun find. `zend_call_function()` bails out early when `EG(exception)` is set, that's why the exception-related tests worked.

Reply:
/cc @joshuaruesweg FYI :point_up_2: